homeassistant/components/wilight/switch.py

```diff
@@ -0,0 +1,393 @@
+)
+
+# Service schemas
+SERVICE_SCHEMA_WATERING_TIME = vol.Schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+        vol.Required(ATTR_WATERING_TIME): VALID_WATERING_TIME,
+    }
+)
+SERVICE_SCHEMA_PAUSE_TIME = vol.Schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+        vol.Required(ATTR_PAUSE_TIME): VALID_PAUSE_TIME,
+    }
+)
+SERVICE_SCHEMA_TRIGGER = vol.Schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+        vol.Required(ATTR_TRIGGER_INDEX): VALID_TRIGGER_INDEX,
+        vol.Required(ATTR_TRIGGER): VALID_TRIGGER,
+    }
+)
+
+# Descriptions of the valve switch entities
+DESC_WATERING = "watering"
+DESC_PAUSE = "pause"
+
+# Icons of the valve switch entities
+ICON_WATERING = "mdi:water"
+ICON_PAUSE = "mdi:pause-circle-outline"
+
+
+def entities_from_discovered_wilight(api_device: Any) -> tuple[Any]:
+    """Parse configuration and add WiLight switch entities."""
+    entities: Any = []
+    for item in api_device.items:
+        if item["type"] == ITEM_SWITCH:
+            index = item["index"]
+            item_name = item["name"]
+            if item["sub_type"] == SWITCH_VALVE:
+                entities.append(WiLightValveSwitch(api_device, index, item_name))
+            elif item["sub_type"] == SWITCH_PAUSE_VALVE:
+                entities.append(WiLightValvePauseSwitch(api_device, index, item_name))
+
+    return entities
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up WiLight switches from a config entry."""
+    parent = hass.data[DOMAIN][entry.entry_id]
+
+    # Handle a discovered WiLight device.
+    entities = entities_from_discovered_wilight(parent.api)
+    async_add_entities(entities)
+
+    # Handle services for a discovered WiLight device.
+    async def set_watering_time(service: Any) -> None:
+        entity_id = service.data[ATTR_ENTITY_ID]
+
+        for entity in entities:
+            if entity.entity_id == entity_id:
+                if isinstance(entity, WiLightValveSwitch):
+                    watering_time = service.data[ATTR_WATERING_TIME]
+                    await entity.async_set_switch_time(watering_time=watering_time)
+
+    async def set_trigger(service: Any) -> None:
+        entity_id = service.data[ATTR_ENTITY_ID]
+
+        for entity in entities:
+            if entity.entity_id == entity_id:
+                if isinstance(entity, WiLightValveSwitch):
+                    trigger_index = service.data[ATTR_TRIGGER_INDEX]
+                    trigger = service.data[ATTR_TRIGGER]
+                    await entity.async_set_trigger(
+                        trigger_index=trigger_index, trigger=trigger
+                    )
+
+    async def set_pause_time(service: Any) -> None:
+        entity_id = service.data[ATTR_ENTITY_ID]
+
+        for entity in entities:
+            if entity.entity_id == entity_id:
```

Comment:
Reverse the condition, continue and outdent below

Reply:
Done "if ((entity.entity_id == entity_id) & isinstance(entity, WiLightValvePauseSwitch)):"